frontend/src/App.tsx

```diff
@@ -20,36 +20,45 @@ import { StudentsList } from "./advising/ManageStudents/StudentsList";
 import { TemplateBuilderPage } from "./advising/Templates/TemplateBuilderPage";
 import { CourseManagmentPage } from "./advising/CourseManagment";
 import { GenericStudentView } from "./advising/ManageStudents/GenericStudentView";
+import ErrorHandler from "./error/ErrorHandler";
 
 export const App = ({ store }: { store: Store }) => {
   return (
     <Provider store={store}>
       <Router>
-        <Switch>
-          {/* requires login */}
-          <ProtectedRoute path="/home" component={HomeWrapper} />
-          <ProtectedRoute path="/redirect" component={RedirectScreen} />
-          <ProtectedRoute path="/onboarding" component={OnboardingInfoScreen} />
-          <ProtectedRoute path="/profile" component={Profile} />
-          <ProtectedRoute path="/management" component={CourseManagmentPage} />
-          <ProtectedRoute
-            path="/completedCourses"
-            component={CompletedCoursesScreen}
-          />
-          <ProtectedRoute
-            path="/transferCourses"
-            component={TransferCoursesScreen}
-          />
-          <ProtectedRoute
-            path="/transferableCredits"
-            component={TransferableCreditScreen}
-          />
-          <ProtectedRoute path="/advisor" component={AdvisorRouter} />
-          {/* requires not logged in */}
-          <UnprotectedRoute path="/" component={Onboarding} />
-          {/* <Route path="/signup" component={SignupScreen} />
-             <Route path="/login" component={LoginScreen} /> */}
-        </Switch>
+        <ErrorHandler>
```

Comment:
The ErrorHandler basically acts as a wrapper that can catch api calls errors such as 401, 404, 500, etc.